app/components/pages/Onboarding.tsx

```diff
@@ -0,0 +1,166 @@
+import { Component } from 'vue-property-decorator';
+import { Onboarding } from 'streamlabs-beaker';
+import TsxComponent from 'components/tsx-component';
+import { OnboardingService } from 'services/onboarding';
+import { Inject } from 'services/core/injector';
+import Connect from './onboarding-steps/Connect';
+import ObsImport from './onboarding-steps/ObsImport';
+import StreamlabsFeatures from './onboarding-steps/StreamlabsFeatures';
+import Optimize from './onboarding-steps/Optimize';
+import FacebookPageCreation from './onboarding-steps/FacebookPageCreation';
+import ThemeSelector from './onboarding-steps/ThemeSelector';
+import { IncrementalRolloutService, EAvailableFeatures } from 'services/incremental-rollout';
+import { UserService } from 'services/user';
+import { $t } from 'services/i18n';
+import styles from './Onboarding.m.less';
+import { throws } from 'assert';
+
+@Component({})
+export default class OnboardingPage extends TsxComponent<{}> {
+  @Inject() onboardingService: OnboardingService;
+  @Inject() incrementalRolloutService: IncrementalRolloutService;
+  @Inject() userService: UserService;
+
+  currentStep = 1;
+  importedFromObs = false;
+  processing = false;
+  fbSetupEnabled: boolean = null;
+
+  stepsState = [{ complete: false }, { complete: false }, { complete: false }];
+
+  checkFbPageEnabled() {
+    if (this.fbSetupEnabled === null || !this.userService.isLoggedIn()) return;
+    // This will do a second unnecessary fetch, but it's the only
+    // way to be sure we have fetched features
+    this.incrementalRolloutService.fetchAvailableFeatures().then(() => {
+      if (this.incrementalRolloutService.featureIsEnabled(EAvailableFeatures.facebookOnboarding)) {
+        this.fbSetupEnabled = true;
+      } else {
+        this.fbSetupEnabled = false;
+      }
+    });
+  }
+
+  continue(importedObs?: boolean) {
+    this.checkFbPageEnabled();
+    this.adjustStepsState(importedObs);
+
+    this.proceed();
+  }
+
+  proceed() {
+    if (this.processing) return;
+    if (this.currentStep === this.stepsState.length) return this.complete();
+
+    this.stepsState[this.currentStep - 1].complete = true;
+    this.currentStep = this.currentStep + 1;
+
+    if (this.currentStep === this.stepsState.length) {
+      this.stepsState[this.currentStep - 1].complete = true;
```

Comment:
Lol - Might have been simpler to make `currentStep` 0-indexed.

Reply:
the beaker component uses a 1-indexed system so i just went with that :P